app_flutter/lib/widgets/task_grid.dart

```diff
@@ -47,12 +49,367 @@ class TaskGridContainer extends StatelessWidget {
         return TaskGrid(
           buildState: buildState,
           commitStatuses: commitStatuses,
+          filterNotifier: filterNotifier,
         );
       },
     );
   }
 }
 
+@immutable
+class TaskGridFilter {
+  TaskGridFilter({
+    RegExp taskFilter,
+    RegExp authorFilter,
+    RegExp messageFilter,
+    this.showAndroid,
```

Comment:
With the ongoing migration to LUCI, this should be clarified to devicelabAndroid along with iOS and windows.

Reply:
I guess I don't understand the configuration of stages and how it relates to the migration. I have left this alone for now while I worked on the other suggestions.

I'm also a little wary of having long property names.

Reply:
Stages are a source of technical debt when there were several different infras powering Flutter. Once everything is moved to LUCI we'll be able to remove the staging logic.